### ISSUE TITLE
Action Modules: Output connected to input

### ISSUE THREAD

**Issue by alexjiao125** [myjeeed/MakerWear#8]
When the output pin of an action module is connected to the output pin of a previous module, it acts like an input pin and the module performs its action.

**Comment by myjeeed** [comment]
This is also causing issues when you're lifting up a module (of any type!) connected to the _output_ of an action module, it would turn it for a short time (like a sparkle).

**Comment by jonfroehlich** [comment]
Yes, this is an unfortunate issue because it affects the user's (child's) mental model of MakerWear, which is both confusing and can impact how they make--two unfortunate consequences.

I believe I previously discussed this with Majeed and we decided that we probably cannot fix this in v1 and for v1 deployments this summer. Majeed can correct me.

**Comment by myjeeed** [comment]
This is true, but as I am becoming more and more eager to do one last round of PCBs, we might be able to fix this.
However, we do need to create a new branch/fork for our next iteration (probably calling it v3) and fix these problems in that version through the Arduino codes as well as the Schematics and Board Layouts.

**Comment by myjeeed** [comment]
And to be more clear, we need to divide the Action Modules **1x3SIN-SOUT** (shown in the attached images, which means signal in and signal out) into two signals: **1x3SIN** and **1x3SOUT**.

<img width="322" alt="screen shot 2016-06-04 at 9 41 19 am" src="https://cloud.githubusercontent.com/assets/9346800/15799848/8543c838-2a38-11e6-815f-29294280ea22.png">
<img width="398" alt="screen shot 2016-06-04 at 9 40 44 am" src="https://cloud.githubusercontent.com/assets/9346800/15799849/854ba166-2a38-11e6-8560-946752bec962.png">

**Comment by jonfroehlich** [comment]
Why v3? I'm confused. Current generation is v1 and we may do another v1 PCB manufacturing run with some hardware fixes and more module designs, right (could call this v1.1)? The next generation architecture that we've discussed (wireless programming, digital rather than analog communication between modules, etc.) is v2, right? So, what's v3 then?

**Comment by myjeeed** [comment]
Whoops! :) yes our previous PCBs would fall under v1.0, our current PCBs will be v1.1 and in order to fix these, we should create v1.2
And yes, the next generation with a new architecture will become v2.

**Comment by jonfroehlich** [comment]
I think that sounds like the right categorization. Thanks for clarifying Majeed!

**Comment by alexjiao125** [comment]
One big problem with this is that once you separate the SIN and SOUT pins, you need to give a PWM signal to the SOUT pin; however in some modules (e.g. the bar graph) the ATtiny does not have a PWM pin available.  We are currently testing to see if we can still connect the SIN and SOUT pins with a diode in between to protect the SOUT pin from acting as an input signal. I think that for modules with an available pin on the ATtiny it should be easier, as we simply connect the PWM pin to the SOUT.
